### ISSUE TITLE
Feature suggestions

### ISSUE THREAD

**Issue by ohyeah521** [erev0s/apkInspector#24]
Hello, I recently saw your open source project, which is very good. I have several functional suggestions: for example, adding functions such as extracting application name, version, mainactivity, packagename, iconfile, etc. These functions are often used in static analysis so that they can be used in UI displays basic information of the application

**Comment by erev0s** [comment]
Hi @ohyeah521, thank you for the suggestions. It is something that has been added in my backlog for quite some time now (https://github.com/androguard/androguard/issues/855#issuecomment-1877843427).
In my little spare time I am working on some new features which I will release in the upcoming months.

**Comment by ohyeah521** [comment]
Thanks for your reply, looking forward to the release of the new version, this will be very useful

**Comment by erev0s** [comment]
@ohyeah521 introduced a fast(lite) parsing of initial element of the AndroidManifest. This returns a dictionary with the relevant elements it reads. The more elements you [define it to read](https://github.com/erev0s/apkInspector/blob/main/apkInspector/axml.py#L887) the slower it will become.

Example:
~~~~
apk_path = '/path/com.airbnb.android_21059188.apk'
parse_apk_for_manifest(apk_path, lite=True)

# output
{'versionCode': '21059188', 'versionName': '20.32', 'installLocation': '0', 'targetSandboxVersion': '2', '': '10', 'package': 'com.airbnb.android', 'platformBuildVersionCode': '29', 'platformBuildVersionName': '10', 'minSdkVersion': '21', 'targetSdkVersion': '28'}
~~~~

**Comment by ohyeah521** [comment]
very practical, can you add an apk application icon extraction function?

**Comment by abirimfo** [comment]
> Hello, I recently saw your open source project, which is very good. I have several functional suggestions: for example, adding functions such as extracting application name, version, mainactivity, packagename, iconfile, etc. These functions are often used in static analysis so that they can be used in UI displays basic information of the application

[za_settings (1).txt](https://github.com/user-attachments/files/16342686/za_settings.1.txt)

**Comment by ohyeah521** [comment]
I hope to add the following function. I wrote a rough function usage for your reference

```
apk = ApkFile(sys.argv[1])  # input apk file path

apk.get_app_name()           # get app label
apk.get_package()              # get package name
apk.get_version()               # get apk version
apk.get_main_activity()      # get main activity

apk.get_manifest()             # get xml format manifest
apk.get_file(file_name)      # get file, eg : b"AndroidManifest.xml"
apk.get_resources(res_id)   # get resource，input resource id，eg: 0x7f100010

apk.get_icon()                    # get icon file path
apk.unzip(out)                   # unzip apk file to out dir
```

**Comment by erev0s** [comment]
thank you for the reference, though please understand the scope of apkInspector. The goal of it is to offer a super simple light weight parser (speed being the important factor) without filtering or fetching from resources. It seems that what you need is a full parser like androguard, to extract the icon and other details.
Currently I do not plan to include this level of detail to apkInspector but maybe its something to consider for the future.

**Comment by ohyeah521** [comment]
Yes, my project integrates androidguard, because I need some detailed information of apk, but androidguard is at a disadvantage in the fight. I paid attention to this project to fight against malware obfuscation a long time ago, but it cannot provide more detailed information at present. Maybe I can implement the encapsulation of these functions after a holiday.

**Comment by erev0s** [comment]
> Yes, my project integrates androidguard, because I need some detailed information of apk, but androidguard is at a disadvantage in the fight. I paid attention to this project to fight against malware obfuscation a long time ago, but it cannot provide more detailed information at present. Maybe I can implement the encapsulation of these functions after a holiday.

FYI we are in the process of "modularization" of androguard.
Ofc we need to find time to progress with it but I believe its a task that I can pick up in the next weeks.
This would allow the axml parser of androguard to be a standalone library that will be used by androguard but also by other tools. apkInspector will be used again to bypass the tricks of malware, and for the axml tricks I will make sure they proper methods will be included in that library the same way I have included them now in androguard.

I think this will be much more beneficial for your project.

**Comment by ohyeah521** [comment]
yes,great job
